src/triage/component/audition/thresholding.py

```diff
@@ -21,6 +22,63 @@ def _of_metric(df, metric_filter):
     ]
 
 
+def model_groups_filter(
+    train_end_times,
+    initial_model_group_ids,
+    models_table,
+    db_engine
+):
+    """Before creating the distance table, we want to filter the model_group_ids
+    and train_end_times only if they are reasonable -- every model group should
+    have the same train_end_times -- to prevent incomparable model groups from
+    being populated to the distance table.
+
+    Args:
+        train_end_times (list) The set of train_end_times to consider during
+                               the iteration
+        initial_model_group_ids (list) The initial list of model group ids to
+                                       narrow down
+        models_table (string) The name of the results schema
+        db_engine (sqlalchemy.engine) A database engine with access to results
+                                      schema of a completed modeling run
+    """
+
+    if isinstance(train_end_times, str) or not hasattr(train_end_times, '__iter__'):
+        raise TypeError("train_end_times should be a list of str or timestamp")
+
+    if not bool(train_end_times):
+        raise ValueError("train_end_times shouldn't be an empty list")
+
+    end_times_sql = "ARRAY[{}]".format(
+        ', '.join(
+            "'{}'".format(
+                end_time.strftime('%Y-%m-%d') if isinstance(end_time, datetime) else end_time
+            ) for end_time in train_end_times
+        )
+    )
+    logging.info("Checking if all model groups have the same train end times")
+    logging.info(f"Found {len(initial_model_group_ids)} total model groups")
```

Comment:
not a problem, but to note, logging generally expects lazy %-interpolation:

    logging.info("Found %s things", len(things))